content/articles/2018-12-readr-1.3.1.Rmarkdown

```diff
@@ -0,0 +1,192 @@
+---
+title: 'readr 1.3.1'
+slug: readr-1-3-1
+description: >
+  readr 1.3.1 is now on CRAN.
+date: '2018-12-21'
+author: Jim Hester
+photo:
+  url: https://unsplash.com/photos/XOW1WqrWNKg
+  author: Anastasia Zhenina
+categories:
+  - package
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+[readr](http://readr.tidyverse.org) 1.3.1 is now available on CRAN! Learn more
+about readr at <http://readr.tidyverse.org>. Detailed notes are always in the
+[change log](https://readr.tidyverse.org/news/index.html#readr-1-3-1).
+
+The readr package makes it easy to get rectangular data out of comma separated
+(csv), tab separated (tsv) or fixed width files (fwf) and into R with code. It
+is designed to flexibly parse many types of data found in the wild, while still
+cleanly failing when data unexpectedly changes. If you are new to readr, the
+best place to start is the data import chapter in R for data science.
+
+The easiest way to install the latest version from CRAN is to install the whole tidyverse.
+
+```{r eval = FALSE}
+install.packages("tidyverse")
+```
+
+Alternatively, install just readr from CRAN:
+
+```{r eval = FALSE}
+install.packages("readr")
+```
+
+readr is part of the core tidyverse, so load it with:
+
+```{r}
+library(tidyverse)
+```
+
+There have been three releases of readr recently, this post will highlight the
+most significant changes across all three releases.
+
+
+## Breaking Changes
+
+### Blank line skipping
+
+The logic for blank line skipping has been overhauled, which should make it
+easier to predict and more robust.
+
+Now both commented lines and completely blank lines are skipped by default.
+
+```{r}
+file <-
+"# A file with comments
+
+first,last
+Mary,Sue
+John,Doe
+"
+
+read_csv(file, comment = "#")
+```
+
+However both commented and blank lines are counted in the `skip` parameter. So
+to skip the header line in this example I would need to use `skip = 3`.
+
+```{r}
+read_csv(file, skip = 3, col_names = FALSE, comment = "#")
+```
```

Comment:
Maybe drop the `comment = "#"` here?

I find these examples somewhat educational too:

``` r
read_csv(file, skip = 1)
#> # A tibble: 2 x 2
#>   first last 
#>   <chr> <chr>
#> 1 Mary  Sue  
#> 2 John  Doe

read_csv(file, skip = 2)
#> # A tibble: 2 x 2
#>   first last 
#>   <chr> <chr>
#> 1 Mary  Sue  
#> 2 John  Doe
```

Because they show that `skip` is a lower bound here on the number of skipped rows.

<sup>Created on 2018-12-21 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>